src/com/yetanalytics/lrs_admin_ui/views/browser.cljs

```diff
@@ -160,61 +161,86 @@
-                    [:ul {:class "action-icon-list"}
-                     [:li
-                      [:a {:href "#!"
-                           :on-click #(dispatch [:browser/clear-filters])
-                           :class "icon-clear-filters"} "Clear Filters"]]]])])]))
-         (if (cstr/blank? content)
-           [:div {:class "browser"}
-            @(subscribe [:lang/get :browser.key-note])]
-           [statement-table {:data content}])]))))
+    (fn [{:keys [label params]}]
+      (when (not-empty params)
+        [:div {:class "filters-wrapper"}
+         [:div {:class (str "pointer collapse-sign"
+                            (when @filter-expand " expanded"))
+                :on-click #(swap! filter-expand not)}
+          label]
+         (when @filter-expand
+           [:div
+            [:div {:class "filter-table"}
+             (map
+              (fn [[key val]]
+                [:div {:class "filter-row"}
+                 [:div {:class "filter-column key"}
+                  key]
+                 [:div {:class "filter-column"}
+                  val]])
+              (seq params))]
+            [:ul {:class "action-icon-list"}
+             [:li
+              [:a {:href ""
+                   :on-click #(dispatch [:browser/clear-filters])
+                   :class "icon-clear-filters"} "Clear Filters"]]]])]))))
+
+(defn- browser-main []
+  (let [content @(subscribe [:browser/get-content])
+        ;; filter out credentials that can't read the LRS
+        read-credentials (filter scopes/has-statement-read-scopes?
+                                 @(subscribe [:db/get-credentials]))]
+    [:div
+     [:p
+      [:span
+       [:b @(subscribe [:lang/get :browser.credentials])]
+       [:select
+        {:name "update_credential"
+         :on-change #(dispatch [:browser/update-credential
+                                (fns/ps-event-val %)])}
+        [:option "Credential to Browse"]
+        (map-indexed
+         (fn [idx credential]
+           [:option {:value (:api-key credential)
+                     :key (str "browser-credential-" idx)}
+            (fns/elide (:api-key credential) 20)])
+         read-credentials)]
+       [refresh-button]]]
+     (let [address @(subscribe [:browser/get-address])
+           params  (httpfn/extract-params address)]
+       (when (some? address)
+         [:div {:class "browser-filters"}
+          [:p @(subscribe [:lang/get :browser.query])]
+          [:div {:class "xapi-address"}
+           address]
+          [filter-view {:label @(subscribe [:lang/get :browser.filters])
+                        :params params}]]))
+     (if (cstr/blank? content)
+       [:div {:class "browser"}
+        @(subscribe [:lang/get :browser.key-note])]
+       [statement-table {:data content}])]))
+
+(defn- csv-download []
+  (let [address @(subscribe [:browser/get-address])
+        params  (httpfn/extract-params address)]
+    [:div
+     [:h4 {:class "content-title"}
```

Comment:
Perhaps add a sentence in normal font here describing CSV Export. Something like:

"You can export your data as a CSV by selecting specific fields as columns below.".

Also maybe just simply make the entire CSV section not appear until a valid cred is selected and data is displayed.

Reply:
About that last point, since we did agree that CSV downloads should be cred-independent, won't doing that make things more confusing?

Reply:
Added instructions in 309095a6fc5bf19092c22660f707e4fba76653f4